techmatters/terraso-mobile-client
# feat: menu components

## Description

[Per design spec](https://www.figma.com/design/KMJbhjVyrrLFHj6jGcX9lN/LandPKS-Design-Prototype-working-file?node-id=7604-45783&m=dev): consolidate the user settings buttons, project user list, and select item lists into a single pair of components called `MenuList` and `MenuItem`. Both are implemented without any direct `NativeBase` dependencies. The new component supports all functionality required by the spec, plus a legacy feature to include the "leave project" button in the user list as a child component. They also support standard accessibility attributes like `accessibilityLabel`, `accessibilityRole`, accessibilityState` etc.

Also refactor some of the `UserList` components to make them easier to reuse now that the ad-hoc menu item implementation has been removed, extracting `ProfilePic` and `RolePill`.

### Checklist

### Related Issues
#1719

### Verification steps

Visit the three affected screens and verify that items are compliant with spec on Figma. I also encourage setting dummy data to check edge-cases like long names, lists that exceed screen height, etc.
